src/frontend/src/components/SearchBar/SearchBar.jsx

```diff
@@ -90,17 +90,26 @@ function CustomizedInputBase(props) {
   const classes = useStyles();
   const { searchText, onChangeHandler, onFilterChangeHandler, filter, onFormSubmit } = props;
 
+  const prepareParamsAndNavigate = (text, filter) => {
+    navigate(`/search?text=${encodeURIComponent(text)}&filter=${encodeURIComponent(filter)}`, {
```

Comment:
Testing this, I think that using navigate is wrong, since it's continuing to accumulate history despite the `replace:true`.  To see what I mean, type something like `react native` and then hit the Back button.  You have to do it for every letter that you typed.

I think we can drop the changes to this file completely, and just use what you have in the SearchPage, since `useQueryParams` lets you set the text and filter values already.

Reply:
Let me clarify this problem. My understanding is when click back button on `http://telescope/search?text=abcde&filter=post`, it should go back to `http://telescope/search/` or other previous history page, not go back to `http://telescope/search?text=abcd&filter=post`. Gatsby navigate( {replace: true}) supposed to do that. But actually, it could not achieve the effect correctly. I also find the problem, but I thought maybe my understanding was wrong. Now I got it. Let me try what else I can do to get there.

Reply:
I tried to remove all navigate and modification from SearchBar.jsx, problem is still there. I think when we use gatsby useQueryParams, it records all typed letter steps to history.